### PR TITLE
[BPK-1396] Finalize phone input

### PR DESCRIPTION
**Pull request by k0nserv**
Adds the docs and finalzes the phone input component.

### CODE REVIEW

**Comment by k0nserv** [comment]
Here's what it looks like

0 0 0 0_8080_components_native_phone-input 2

**Comment by k0nserv** [comment]
Here's what the fully integrated example


phone-input

**Comment by matthewdavidson** [comment]
Can we add placeholders to all examples please :)

File: native/packages/react-native-bpk-component-phone-input/stories.js [hunk 1]
```diff
@@ -141,11 +165,87 @@ class StatefulBpkDialingCodeList extends React.Component<
...
+  'Full component example',
+  () => <FullyIntegrated initiallySelectedId="CA" codes={codes} />,
+);
+
+storiesOf('phone-input/BpkDialingCodeList', module).add('docs:default', () => (
   <StatefulBpkDialingCodeList />
 ));
 
-storiesOf('BpkDialingCodeListItem', module)
+storiesOf('phone-input/BpkDialingCodeListItem', module)
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-phone-input/stories.js]: This is inconsistent with the rest of the story names.

File: packages/bpk-docs/src/pages/NativePhoneInputPage/NativePhoneInputPage.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  },
+];
+
+const NativeInputPage = () => (
+  <DocsPageBuilder
+    title="Phone number input"
+    blurb={[
+      <Paragraph>
+        The Backpack phone number input package,
+        <BpkCode>react-native-bpk-component-phone-input</BpkCode>, contains two
```
Comment (matthewdavidson) [inline packages/bpk-docs/src/pages/NativePhoneInputPage/NativePhoneInputPage.js]: remove the full package name here , its inconsistent with what we do on other pages.

File: packages/bpk-docs/src/pages/NativePhoneInputPage/NativePhoneInputPage.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        components that together enable the collection of phone numbers. <br />These
+        are:
+        <BpkList>
+          <BpkListItem>A dialing code list component.</BpkListItem>
+          <BpkListItem>A specialized text input field.</BpkListItem>
+        </BpkList>
+        Consumers are responsible for using these two components together. The
+        accessorView in the text input shall present the dialing code list when
```
Comment (matthewdavidson) [inline packages/bpk-docs/src/pages/NativePhoneInputPage/NativePhoneInputPage.js]: use english here: i.e. just "accessory view"

**Comment by k0nserv** [comment]
Before we finalize this are we all okay with the API and terminonlogy? Is _codes_ the right word to use in props for example? Is the word data in `dialingCodeData` redundant?

**Comment by shaundon** [comment]
> Is codes the right word to use in props for example

I know it was me that named it `codes` in the first place, but now I'm thinking it's too generic sounding. How about `diallingCodes`?

**Comment by matthewdavidson** [comment]
@shaundon, @k0nserv I opt for `data` in both instances.

**Comment by k0nserv** [comment]
So `dialingCodes` in `BpkDialingCodeList` and `dialingCode` in `BpkPhoneNumberInput`?

> @shaundon, @k0nserv I opt for data in both instances.

In the calendar we use a more descriptive name, I think we should do the same here

**Comment by shaundon** [comment]
> So dialingCodes in BpkDialingCodeList and dialingCode in BpkPhoneNumberInput?

I like that!

**Comment by k0nserv** [comment]
Cool. I've made those chagnes now. Please review

File: native/packages/react-native-bpk-component-animate-height/stories.js [hunk 4]
```diff
@@ -102,7 +102,7 @@ AnimateHeightDemo.defaultProps = {
   animationDuration: animationDurationBase,
 };
 
-storiesOf('BpkAnimateHeight', module)
+storiesOf('react-native-bpk-component-antimate-height', module)
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-animate-height/stories.js]: `antimate` ➡️ `animate`

File: native/packages/react-native-bpk-component-phone-input/readme.md [hunk 5]
```diff
@@ -8,6 +8,20 @@
 npm install react-native-bpk-component-phone-input --save-dev
 ```
 
+## Data format
+
+Consumers are expected to provide the data that powers the two components available in this package.
+
+Each supported country should be in the following format:
+
+```javscript
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-phone-input/readme.md]: `javscript` ➡️ `javascript`

File: native/packages/react-native-bpk-component-touchable-native-feedback/src/BpkTouchableNativeFeedback.js [hunk 6]
```diff
@@ -50,7 +50,7 @@ const BpkTouchableNativeFeedback = (props: Props) => {
 
 BpkTouchableNativeFeedback.propTypes = {
   children: PropTypes.element.isRequired,
-  borderlessBackground: PropTypes.boolean,
+  borderlessBackground: PropTypes.bool,
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-touchable-native-feedback/src/BpkTouchableNativeFeedback.js]: I keep seeing an error in storybooks to do with this - I expect this line will fix it! Nice one 👍

File: packages/bpk-docs/src/pages/NativePhoneInputPage/NativePhoneInputPage.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    blurb={[
+      <Paragraph>
+        The Backpack phone number input package contains two components that
+        together enable the collection of phone numbers. <br />These are:
+        <BpkList>
+          <BpkListItem>A dialing code list component.</BpkListItem>
+          <BpkListItem>A specialized text input field.</BpkListItem>
+        </BpkList>
+        Consumers are responsible for using these two components together. The
+        accessor view in the text input shall present the dialing code list when
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/NativePhoneInputPage/NativePhoneInputPage.js]: `accessor` ➡️ `accessory`

**Comment by k0nserv** [comment]
> The other day, we talked about exposing keyboardType but setting it to phone-pad by default, so that consumers can change it if they need to.

It is in there becuase consumers can still override it

**Comment by shaundon** [comment]
> It is in there becuase consumers can still override it

Ah yeah, I guess they can do it through `{...rest}`.

**Comment by k0nserv** [comment]
Comments addressed

**Pull request merged**